xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1455,10 +1454,20 @@ void CGUIMediaWindow::OnInitWindow()
 
   m_rootDir.SetAllowThreads(true);
 
-  if (m_iSelectedItem > -1)
-    m_viewControl.SetSelectedItem(m_iSelectedItem);
-
   CGUIWindow::OnInitWindow();
+
+  m_viewControl.SetSelectedItem(GetMediaSelectedItem(m_vecItems->GetPath()));
```

Comment:
do you mean with m_viewControl.SetSelectedItem? it need always to get only the item position

it's SetMediaSelectedItem that takes 2 argument, the current  path and the item position